model/patch/patch.go

```diff
@@ -152,6 +160,7 @@ type Patch struct {
 	BackportOf      BackportInfo           `bson:"backport_of,omitempty"`
 	MergePatch      string                 `bson:"merge_patch"`
 	GithubPatchData thirdparty.GithubPatch `bson:"github_patch_data,omitempty"`
+	GitInfo         GitMetadata            `bson:"git_info"`
```

Comment:
Can this be omitempty? Most patches won't have it

Reply:
I've made the fields of GitMetadata omitempty, but [I don't think we'd gain by setting this field omitempty](https://github.com/evergreen-ci/evergreen/pull/4093).

Reply:
But can it be made a pointer?

Reply:
that works. done
I think it's better this way for another reason too: it's easier to disambiguate when it hasn't been provided by the old CLI.